internal/rbd/replicationcontrollerserver.go

```diff
@@ -445,20 +436,16 @@ func (rs *ReplicationServer) ResyncVolume(ctx context.Context,
 	}
 	ready := false
 	state := imageMirroringState(mirrorStatus.State)
-	if state == upAndStopped || state == upAndReplaying {
-		// Make sure the peer site image state is up and stopped
-		ready = true
-		for _, s := range mirrorStatus.PeerSites {
-			if imageMirroringState(s.State) != upAndStopped {
-				util.UsefulLog(ctx, "peer site name=%s, mirroring state=%s, description=%s and lastUpdate=%s", s.SiteName, s.State, s.Description, s.LastUpdate)
-				ready = false
-			}
-		}
-	}
-
-	// when the images are demoted on both clusters and user requests for the
-	// resync of the image, the image mirror state will be unknown state in
-	// both clusters.
+	//  To recover from split brain (up+error) state the image need to be
+	//  demoted and requested for resync on site-a and then the image on site-b
+	//  should gets demoted.The volume should be marked to ready=true when the
+	//  image state on both the clusters are up+unknown because during the last
+	//  snapshot syncing the data gets copied first and then image state on the
+	//  site-a changes to up+unknown.
```

Comment:
Hmm, I do not understand how this is done. Does the admin need to do this manually? There probably should be some logging for this case?

Reply:
Yes, Failover and failback are always admin triggered actions. we are already logging the image state when resync is not completed or its going on. IMO it should be enough for now.

Reply:
Ok, thanks for explaining